index.js

```diff
@@ -217,15 +230,19 @@ function adapter(uri, opts){
 
     debug('removing %s from rooms %s', id, Object.keys(rooms));
 
-    Adapter.prototype.delAll.call(this, id);
```

Comment:
Your implementation of delAll() function don't check that any other sockets leave in the room.
https://github.com/GannettDigital/socket.io-ioredis/blob/master/index.js#L226
The situation of issue is written in top of this conversation. 
Adapter's delAll() function is remove only socketid from Room instance.
https://github.com/socketio/socket.io-adapter/blob/master/index.js#L85
Before unsubscribe I have to check that no sockets are left in the room.
This logic is as same as del() function.
https://github.com/GannettDigital/socket.io-ioredis/pull/2/files#diff-168726dbe96b3ce427e7fedce31bb0bcR240
https://github.com/GannettDigital/socket.io-ioredis/pull/2/files#diff-168726dbe96b3ce427e7fedce31bb0bcR204

My English is poor, but I appreciate your kindness.